### PR TITLE
Feature/onboarding

### PR DESCRIPTION
**Pull request by DmitrySuprun**
https://user-images.githubusercontent.com/95674666/195723120-3cafef07-138c-4834-9d7c-e3b55157fbeb.mov

### CODE REVIEW

File: AppleStore/AppleStore/SceneDelegate.swift [hunk 1]
```diff
@@ -17,7 +17,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                options connectionOptions: UIScene.ConnectionOptions) {
         guard let windowScene = (scene as? UIWindowScene) else { return }
         window = UIWindow(windowScene: windowScene)
-        let rootViewController = StoreTabBarViewController()
+//        let rootViewController = StoreTabBarViewController()
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/SceneDelegate.swift:20]:
```suggestion
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+//  OnboardingItemViewController.swift
+//  AppleStore
+//
+//  Created by Дмитрий Супрун on 13.10.22.
+//
+
+import UIKit
+
+/// Current page for OnboardingPageViewController
+class OnboardingItemViewController: UIViewController {
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift:11]:
```suggestion
final class OnboardingItemViewController: UIViewController {
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    
+    // MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        UIView.animate(withDuration: 1, delay: 0) {
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift:67]: анимацию в метод

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    
+    // MARK: - Objc Private Methods
+    @objc private func skipActon() {
+        let nextViewController = StoreTabBarViewController()
+        nextViewController.modalPresentationStyle = .fullScreen
+        present(nextViewController, animated: true)
+    }
+    
+    @objc private func nextAction() {
+        guard let currentViewController = viewControllers?.first else { return }
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:96]: в один guard

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    
+    private func addViews() {
+        view.addSubview(skipButton)
+        view.addSubview(nextButton)
+        view.addSubview(getStarted)
+    }
+    
+    private func checkIndex() {
+        for subview in self.view.subviews where subview is UIPageControl {
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:126]:
```suggestion
        for subview in view.subviews where subview is UIPageControl {
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    }
+}
+
+// MARK: - UIPageViewControllerDataSource
+extension OnboardingPageViewController: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController,
+                            viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        guard let viewController = viewController as? OnboardingItemViewController else { return nil }
+        if let index = pagesViewControllers.firstIndex(of: viewController) {
+            print(index)
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:151]:
```suggestion
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        }
+        userDefaults.set(false, forKey: Constants.isFirstStartKey)
+    }
+}
+
+// MARK: - UIPageViewControllerDataSource
+extension OnboardingPageViewController: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController,
+                            viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        guard let viewController = viewController as? OnboardingItemViewController else { return nil }
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:149]:
```suggestion
        guard 
            let viewController = viewController as? OnboardingItemViewController, 
            if index > 0 
        else {
           return nil
           }
           
       for subview in view.subviews where subview is UIPageControl {
              subview.isHidden = false
              getStarted.isHidden = true
              nextButton.isHidden = false
              skipButton.isHidden = false
            }
        return pagesViewControllers[index - 1] 
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+            }
+        }
+        return nil
+    }
+
+    func pageViewController(_ pageViewController: UIPageViewController,
+                            viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        guard let viewController = viewController as? OnboardingItemViewController else { return nil }
+        if let index = pagesViewControllers.firstIndex(of: viewController) {
+            print(index)
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:169]:
```suggestion
```

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+            if index == pagesViewControllers.count - 1 {
+                for subview in self.view.subviews where subview is UIPageControl {
```
Comment (AlexanderKokh) [inline AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:171]:
```suggestion
                for subview in view.subviews where subview is UIPageControl {
```

File: AppleStore/AppleStore/SceneDelegate.swift (hunk 1 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/SceneDelegate.swift:20]: Deletted

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift (hunk 2 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift:11]: Fixed

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift (hunk 3 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingItemViewController.swift:67]: Fixed

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 4 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:96]: Fixed

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 5 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:126]: Fixed

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 6 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:151]: Deleted

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 7 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:149]: -

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 8 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:169]: Deleted

File: AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift (hunk 9 above)
Comment (DmitrySuprun) [reply AppleStore/AppleStore/ViewControllers/Onboard/OnboardingPageViewController.swift:171]: Deleted

**Pull request merged**